### PR TITLE
[ESKL-351] SkillsViewModel blocking operations refactor

### PR DESCRIPTION
**Pull request by TiagoAVieira**
**What does this PR do?**

   Refactors SkillsViewModel to remove blocking operations. Additionally, it is an attempt to clean some of the logic inside SkillsFragment to make it more manageable. Future refactoring will be necessary, but this is a first step and a noticeable improvement.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] legacy/skills/src/main/java/cm/aptoide/skills/SkillsViewModel.kt

**How should this be manually tested?**

  Test normal eskills flow and referral flow.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ESKL-351](https://aptoide.atlassian.net/browse/ESKL-351)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

[ESKL-351]: https://aptoide.atlassian.net/browse/ESKL-351?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rodolfobcardoso — CHANGES_REQUESTED** [review]
Just as a note (no need to change this), usually the .subscribe is done in the viewModel and not in the fragment. The fragment is reconstructed when you rotate the screen or minimize the app, so the rx chains will be created again in this cases.
Since the subscribe is in the fragment, I recommend testing rotating the screen in the middle of the flow to be sure nothing is breaking

File: legacy/skills/src/main/java/cm/aptoide/skills/SkillsFragment.kt [hunk 1]
```diff
@@ -201,44 +211,51 @@ class SkillsFragment : Fragment(), PaymentView {
...
-              views.onboardingLayout.dialogBuyButtonsPaymentMethods.buyButton.isEnabled = true
-            }
-
-            is SuccessfulReferral -> {
-              views.onboardingLayout.root.visibility = View.GONE
-              createAndPayTicket(eSkillsPaymentData, true)
-              analytics.sendOnboardingSuccessEvent(eSkillsPaymentData, referralCode)
-            }
+                }
+              }.subscribe()
```
[the remark this answers is not in the document]
Comment (TiagoAVieira) [reply legacy/skills/src/main/java/cm/aptoide/skills/SkillsFragment.kt:256]: Got it, will do.

[the remark this answers is not in the document]
Comment (TiagoAVieira) [reply legacy/skills/src/main/java/cm/aptoide/skills/SkillsFragment.kt]: Fixed

File: legacy/skills/src/main/java/cm/aptoide/skills/SkillsFragment.kt [hunk 2]
```diff
@@ -379,7 +401,8 @@ class SkillsFragment : Fragment(), PaymentView {
         details.fiatPrice.text = "${it.amount} ${it.currency}"
         details.fiatPriceSkeleton.visibility = View.GONE
         details.fiatPrice.visibility = View.VISIBLE
-      }.subscribe(), viewModel.getFormattedAppcAmount(
+      }.subscribe(),
```
[the remark this answers is not in the document]
Comment (TiagoAVieira) [reply legacy/skills/src/main/java/cm/aptoide/skills/SkillsFragment.kt]: Added onError

**Review by rodolfobcardoso — APPROVED** [review]

**Review by rodolfobcardoso — APPROVED** [review]

**Pull request closed without merging**